.github/workflows/build.yml

```diff
@@ -36,9 +37,35 @@ concurrency:
 env:
   MAVEN_ARGS: "-e -B --settings .github/mvn-settings.xml --fail-at-end"
 jobs:
+  quick-build:
+    name: "Initial JDK Build"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Generate .m2 cache key
+        id: m2-cache-key
+        run: |
+          echo "key=m2-cache-$(/bin/date -u "+%Y-%U")" >> $GITHUB_OUTPUT
+      - name: Cache Maven local repository
+        id: cache-maven
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          # refresh cache every week to avoid unlimited growth
+          key: ${{ steps.m2-cache-key.outputs.key }}
+      - name: Building code and running basic checks
+        run: ./mvnw $MAVEN_ARGS install -Pjqassistant -Pdist -Pci-sources-check -DskipTests
+      - name: Delete snapshots artifacts from cache
+        run: find ~/.m2 -name \*-SNAPSHOT -type d -exec rm -rf {} +
```

Comment:
This means some unit tests won't run on Windows though, doesn't it? In particular the ones you're working on right now, those about CodeSource...

Also if we just cache the local maven repo, not the `target` directories, I believe the second, incremental build will rebuild some artifacts (though without running unit tests). So the "quick build" really brings very little if we do it this way.

Can't we keep everything in the same job except the "date" thing? Or find a different way of computing the date on Windows?

Reply:
What if we go with the example from the action docs: https://github.com/actions/cache/blob/main/examples.md#java---maven
this seems reasonable, if the pom files changed the key would change ...